src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -383,196 +411,318 @@ public HttpResponseMessage PrintPdf(PostedDataEntity loadDocumentRequest)
-            {
-                // Set watermark properties
-                watermark = new Watermark(globalConfiguration.Viewer.GetWatermarkText());
-                watermark.Color = System.Drawing.Color.Blue;
-                watermark.Position = WatermarkPosition.Diagonal;
-                watermark.Width = 100;
-            }
-            options.EmbedResources = true;
-            // set password for protected document
-            if (!string.IsNullOrEmpty(password))
-            {
-                options.Password = password;
-            }
-            if (watermark != null)
+            var xdoc = new XDocument(new XElement("Pages"));
+
+            foreach (var page in viewInfo.Pages)
             {
-                options.Watermark = watermark;
+                xdoc.Element("Pages")
+                    .Add(new XElement("PageData",
+                        new XElement("Number", page.Number),
+                        new XElement("Angle", 0)));
             }
-            if (pageNumber != 0)
+
+            xdoc.Save(pagesInfoPath);
+        }
+
+        private int GetCurrentAngle(int pageNumber, string pagesInfoPath)
+        {
+            XDocument xdoc = XDocument.Load(pagesInfoPath);
+            var pageData = xdoc.Descendants()?.Elements("Number")?.Where(x => int.Parse(x.Value) == pageNumber)?.Ancestors("PageData");
+            var angle = pageData.Elements("Angle").FirstOrDefault();
+
+            if (angle != null)
             {
-                options.PageNumber = pageNumber;
-                options.CountPagesToRender = 1;
+                return int.Parse(angle.Value);
             }
-            options.CellsOptions.ShowGridLines = globalConfiguration.Viewer.GetShowGridLines();
+
+            return 0;
         }
 
-        private void SetOptions(ImageOptions options, string password, int pageNumber)
+        private void SetWatermarkOptions(ViewOptions options)
         {
             Watermark watermark = null;
-            if (!String.IsNullOrEmpty(globalConfiguration.Viewer.GetWatermarkText()))
+            if (!string.IsNullOrEmpty(globalConfiguration.Viewer.GetWatermarkText()))
             {
                 // Set watermark properties
                 watermark = new Watermark(globalConfiguration.Viewer.GetWatermarkText());
                 watermark.Color = System.Drawing.Color.Blue;
-                watermark.Position = WatermarkPosition.Diagonal;
-                watermark.Width = 100;
-            }
-            // set password for protected document
-            if (!string.IsNullOrEmpty(password))
-            {
-                options.Password = password;
+                watermark.Position = Position.Diagonal;
             }
+
             if (watermark != null)
             {
                 options.Watermark = watermark;
             }
-            if (pageNumber != 0)
+        }
+
+        private Options.LoadOptions GetLoadOptions(string password)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'GetLoadOptions' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5242614)